resources/lib/SUBUtilities.py

```diff
@@ -260,3 +266,58 @@ def request(self, url, data=None, query_string=None, referrer=None, cookie=None)
             log("Failed to get url: %s\n%s" % (url, e))
             # Second parameter is the filename
         return content
+
+
+def title_from_focused_item(item_data):
+    label_type = xbmc.getInfoLabel("ListItem.DBTYPE")  # movie/tvshow/season/episode
+    label_movie_title = xbmc.getInfoLabel("ListItem.OriginalTitle")
+    is_movie = xbmc.getCondVisibility("Container.Content(movies)") or label_type == 'movie'
+    is_episode = xbmc.getCondVisibility("Container.Content(episodes)") or label_type == 'episode'
+
+    title = ''
+    if is_movie and label_movie_title and item_data['year']:
+        title = label_movie_title + " " + item_data['year']
+    elif is_episode and item_data['tvshow'] and item_data['season'] and item_data['episode']:
+        title = ("%s S%.2dE%.2d" % (item_data['tvshow'], int(item_data['season']), int(item_data['episode'])))
+
+    return title
+
+def getTVshowOriginalTitleByTMDBapi():
+    labelTVShowTitle = xbmc.getInfoLabel("ListItem.TVShowTitle")
+    labelYear = xbmc.getInfoLabel("ListItem.Year")
+    tmdbKey = '653bb8af90162bd98fc7ee32bcbbfb3d'
+    filename = 'screwzira.search.tmdb.%s.%s.%s.json' % ("tv",lowercase_with_underscores(labelTVShowTitle), labelYear)
+    if labelYear > 0:
+        url = "http://api.tmdb.org/3/search/%s?api_key=%s&query=%s&year=%s&language=en" % (
+			"tv",tmdbKey, labelTVShowTitle, labelYear)
+    else:
+        url = "http://api.tmdb.org/3/search/%s?api_key=%s&query=%s&language=en" % (
+			"tv",tmdbKey, labelTVShowTitle)
+    
+    log("searchTMDB: %s" % url)
+    json = cachingJSON(filename,url)
```

Comment:
What is the purpose of this cache?

Reply:
Basically this code was taken from another place.
But as I understand it saves communication with the tmdb api if you've already converted to the same series name before, because the json data file already exist in the addon data folder.